### PR TITLE
feat: add Dropdown field

### PR DESCRIPTION
**Pull request by smhigley**
We had ComboboxField, but DropdownField was missing 😅

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/Avatar-mount_base.html">1526</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/Avatar-mount.html">1552</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/Button-mount_base.html">1117</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/Button-mount.html">1111</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/FluentProvider-mount_base.html">1912</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/FluentProvider-mount.html">1952</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/FluentProviderWithTheme-mount_base.html">749</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/FluentProviderWithTheme-mount.html">747</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">714</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">712</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">765</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">755</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/InfoButton-mount_base.html">687</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/InfoButton-mount.html">670</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/MakeStyles-mount_base.html">2259</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/MakeStyles-mount.html">2257</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/Persona-mount_base.html">3624</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/Persona-mount.html">3557</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/SpinButton-mount_base.html">3025</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26488/perf-test-react-components/SpinButton-mount.html">2984</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                          | Baseline (minified/GZIP) |                           PR |       Change |
| :--------------------------------------------------------------------------------------------------------- | -----------------------: | ---------------------------: | -----------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/DropdownField.fixture.js'>DropdownField</abbr> |         `0 B`<br />`0 B` | `77.447 kB`<br />`25.849 kB` | 🆕 New entry |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                             |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr> | `80.881 kB`<br />`25.957 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/ComboboxField.fixture.js'>ComboboxField</abbr>                    | `78.763 kB`<br />`26.095 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr> | `79.549 kB`<br />`25.686 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/4f3871db826e542114d853b57fb2f740787f91e8'>4f3871db826e542114d853b57fb2f740787f91e8</a></sub>
<div id="bundleSizeComment9423"/>

File: apps/vr-tests-react-components/src/stories/Field.stories.tsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { Steps, StoryWright } from 'storywright';
 import { storiesOf } from '@storybook/react';
```
Comment (fabricteam) [inline apps/vr-tests-react-components/src/stories/Field.stories.tsx:3]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 7 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=26488&commitId=48471804cba82ab823294dc907622517b76f9494&env=prod)

✅ There was 7 screenshots added, 0 screenshots removed, 1775 screenshots unchanged, 0 screenshots with different dimensions and 0 screenshots with visible difference.


  <details><summary><b>unknown</b> 7 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|DropdownField converged.base.default.chromium.png|0| Added|
|DropdownField converged.hint.default.chromium.png|0| Added|
|DropdownField converged.horizontal.default.chromium.png|0| Added|
|DropdownField converged.required.default.chromium.png|0| Added|
|DropdownField converged.size-large.default.chromium.png|0| Added|
|DropdownField converged.size-small.default.chromium.png|0| Added|
|DropdownField converged.validation.default.chromium.png|0| Added|

</details>


<div id="vrtCommentfluentuiv9"/>

**Comment by smhigley** [comment]
Closing, since #26430 is almost ready to be merged, and will make ComboboxField/DropdownField obsolete

**Pull request closed without merging**